lib/asciidoctor-revealjs/highlightjs.rb

```diff
@@ -11,10 +11,20 @@ def initialize *args
           @name = @pre_class = 'highlightjs'
         end
 
+        # Convert between slide notation formats
+        # Asciidoctor uses 1..3,6,8 whereas reveal.js expects 1-3,6,8
+        def _convert_linedef_to_revealjs highlight_lines
+          return highlight_lines.gsub("..", "-")
+        end
+
         def format node, lang, opts
           super node, lang, (opts.merge transform: proc { |_, code|
             code['class'] = %(language-#{lang || 'none'} hljs)
             code['data-noescape'] = true
+            # Note for review: should the API be modified to give easier access to this?
+            if node.attributes.key?("highlight")
+              code['data-line-numbers'] = self._convert_linedef_to_revealjs(node.attributes["highlight"])
+            end
```

Comment:
I don't think my access to the `highlight` attribute of the listing block is proper here. It should go through the syntax highlight API but I didn't find a way to do so. `opts[:highlight_lines]` is available in `highlight()` but not in `format()`.

@Mogztter, @mojavelinux thoughts?

Reply:
Sorry I can't help, I'm not familiar with this API so I don't know the reason behind this choice

Reply:
I didn't think about the fact that the option would be needed in format. We could take that up in core. In the meantime, you can call:

```
resolve_lines_to_highlight source, (attr 'highlight')
```

Reply:
don't do it this way:

```
highlight_lines.gsub("..", "-")
```

The logic is much more complex than that. See https://github.com/asciidoctor/asciidoctor/blob/master/lib/asciidoctor/substitutors.rb#L968-L999

Reply:
The problem is that reveal.js introduced a new legal character `|` to split highlights into steps sequences. It's a very cool feature that I intend to use in presentations. Documented here: https://github.com/hakimel/reveal.js#step-by-step-highlights

So I'm going to need to split on it, then send each section to `resolve_lines_to_highlight` method and rejoin everything after. I'm on it right now.

In the meantime, here how I adapted your code to make it work, in case you have any comment:

```ruby
if node.attributes.key?("highlight")
    code['data-line-numbers'] = node.resolve_lines_to_highlight(node.content, node.attributes["highlight"]).join(',')
```

Reply:
Fixed in a839f53. This is probably the most idiomatic ruby I've written so far. Marking this as resolved but allowing you to further comment for another day before merging.